Author: Michael Palimaka <kensington@gentoo.org>
Description: Fix build with V4L and recent kernels.
 .
 REVIEW: 7072
 BUG: 319455
Applied-Upstream: 4.10.4, commit:1353735

Index: kdenetwork/kopete/libkopete/avdevice/videodevice.cpp
===================================================================
--- kdenetwork/kopete/libkopete/avdevice/videodevice.cpp	(revision 1353734)
+++ kdenetwork/kopete/libkopete/avdevice/videodevice.cpp	(revision 1353735)
@@ -3050,7 +3050,9 @@
 		case V4L2_CID_AUDIO_TREBLE: 		return I18N_NOOP("Treble");
 		case V4L2_CID_AUDIO_MUTE: 		return I18N_NOOP("Mute");
 		case V4L2_CID_AUDIO_LOUDNESS: 		return I18N_NOOP("Loudness");
+#ifdef V4L2_CID_BLACK_LEVEL		// deprecated, likely to be removed soon
 		case V4L2_CID_BLACK_LEVEL:		return I18N_NOOP("Black Level");
+#endif
 		case V4L2_CID_AUTO_WHITE_BALANCE:	return I18N_NOOP("Automatic White Balance");
 		case V4L2_CID_DO_WHITE_BALANCE:		return I18N_NOOP("Do White Balance");
 		case V4L2_CID_RED_BALANCE:		return I18N_NOOP("Red Balance");
@@ -3061,8 +3063,10 @@
 		case V4L2_CID_GAIN:			return I18N_NOOP("Gain");
 		case V4L2_CID_HFLIP:			return I18N_NOOP("Horizontal Flip");
 		case V4L2_CID_VFLIP:			return I18N_NOOP("Vertical Flip");
+#ifdef V4L2_CID_HCENTER			// removed in kernel 3.9
 		case V4L2_CID_HCENTER:			return I18N_NOOP("Horizontal Center");
 		case V4L2_CID_VCENTER:			return I18N_NOOP("Vertical Center");
+#endif
 #ifdef V4L2_CID_POWER_LINE_FREQUENCY	// since kernel 2.6.25
 		case V4L2_CID_POWER_LINE_FREQUENCY:	return I18N_NOOP("Power Line Frequency");
 		case V4L2_CID_HUE_AUTO:			return I18N_NOOP("Automatic Hue");
